packages/tx-construction/src/tx-builder/TxBuilder.ts

```diff
@@ -246,15 +248,28 @@ export class GenericTxBuilder implements TxBuilder {
   }
 
   async #getOrCreateRewardAccounts(): Promise<RewardAccountWithPoolId[]> {
+    let newRewardAccounts: Cardano.RewardAccount[] = [];
     if (this.#requestedPortfolio) {
-      await util.ensureStakeKeys({
+      newRewardAccounts = await util.ensureStakeKeys({
         count: this.#requestedPortfolio.length,
         keyAgent: this.#dependencies.keyAgent,
         logger: contextLogger(this.#logger, 'getOrCreateRewardAccounts')
       });
     }
 
-    return this.#dependencies.txBuilderProviders.rewardAccounts();
+    const rewardAccounts$ = coldObservableProvider({
+      pollUntil: (rewardAccounts) =>
+        newRewardAccounts.every((newAccount) => rewardAccounts.some((acct) => acct.address === newAccount)),
+      provider: this.#dependencies.txBuilderProviders.rewardAccounts,
+      // TODO: find some good values for the retries
+      retryBackoffConfig: { initialInterval: 100, maxInterval: 200, maxRetries: 10 }
```

Comment:
I think this API is confusing because I expected there to be a `trigger$` to control the polling. Now there are 2 ways to do the polling :sweat_smile: Not in scope if this PR though.
```suggestion
      trigger$: interval(100)
```

I suggest `{ initialInterval: 10, maxInterval: 100, maxRetries: 10 }` but that's not based on anything